### PR TITLE
JSON/ZLib file formats - v1.6

### PR DESCRIPTION
**Pull request by davidfig**
- [x] Upgrade to v1.6 file format, supporting bincode serialization
- [x] Serialize with bincode then compress with zilib on export
- [x] Decompress with zlib and deserialize with bincode on import 
- [x] Update quadratic-client and quadratic-api to support binary files
- [x] Add file version to the exported binary
- [x] Remove file version from the imported binary
- [x] Auto generate a current_blank.grid file and add to package.json
- [x] Use already created current_blank.grid file for new files (bypass the client)
- [x] Remove unused legacy file-related code in quadratic-client

### CODE REVIEW

**Comment by davidfig** [comment]
Code LGTM . I can't approve since I started this PR. @ddimaria please approve.

**Review by davidkircos — CHANGES_REQUESTED** [review]
Importing this `.grid` file from the dashboard fails with rust panic

https://fastupload.io/688a351c18be86b3

File: quadratic-core/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,10 @@ path = "src/bin/docgen.rs"
 name = "export_types"
 path = "src/bin/export_types.rs"
 
+[[bin]]
+name = "generate_blank_current_file"
```
[the remark this answers is not in the document]
Comment (ddimaria) [reply quadratic-core/Cargo.toml:24]: This automates generating and placing a blank grid file of the current version to be used in 2 places in the repo.

**Comment by ddimaria** [comment]
> Importing this `.grid` file from the dashboard fails with rust panic
> 
> https://fastupload.io/688a351c18be86b3

This is fixed and related to new upstream code (rows_resize)

**Review by ddimaria — APPROVED** [review]

**Pull request merged**